CoreSchemaView/src/au/gov/asd/tac/constellation/views/schemaview/providers/VertexTypeNodeProvider.java

```diff
@@ -223,79 +223,41 @@ private void populateTree() {
-        boolean found = false;
-        final String name = treeItem.getName();
-        found = isFilterMatchText(name);
-        if (!found) {
-            final String description = treeItem.getDescription();
-            found = isFilterMatchText(description);
-        }
-        if (!found) {
-            final String color = treeItem.getColor().toString();
-            found = isFilterMatchText(color);
-        }
-        if (!found) {
-            final String foregroundIconLabel = treeItem.getForegroundIcon().getName();
-            found = isFilterMatchText(foregroundIconLabel);
-        }
-        if (!found) {
-            if (treeItem.getValidationRegex() != null) {
-                final String validationLabel = treeItem.getValidationRegex().toString();
-                found = isFilterMatchText(validationLabel);
-            }
-        }
-        if (!found) {
-            if (treeItem.getDetectionRegex() != null) {
-                final String detectionLabel = treeItem.getDetectionRegex().toString();
-                found = isFilterMatchText(detectionLabel);
-            }
-        }
-        if (!found) {
-            final String hierachy = treeItem.getHierachy();
-            found = isFilterMatchText(hierachy);
-        }
-        if (!found) {
-            for (String property : treeItem.getProperties().keySet()) {
-                final Object propertyValue = treeItem.getProperty(property);
-                if (propertyValue != null) {
-                    found = isFilterMatchText(propertyValue.toString());
-                    if (found) {
-                        break;
-                    }
-                }
-            }
-        }
-        return found;
+    private boolean isFilterMatchAnyProperty(SchemaVertexType treeItem) {
+        return isFilterMatchText(treeItem.getName())
+                || isFilterMatchText(treeItem.getDescription())
+                || isFilterMatchText(treeItem.getColor().getName())
+                || isFilterMatchText(treeItem.getForegroundIcon().getName())
+                || isFilterMatchText(Objects.toString(treeItem.getValidationRegex(), ""))
+                || isFilterMatchText(Objects.toString(treeItem.getDetectionRegex(), ""))
+                || isFilterMatchText(treeItem.getHierachy())
+                || !(treeItem.getProperties().keySet().isEmpty()) && treeItem.getProperties().keySet().stream().anyMatch(property
+                -> property != null && isFilterMatchText(property.toString())
+        );
     }
 
     private boolean isFilterMatchText(final String propertyValue) {
         final String filterInputText = filterText.getText().toLowerCase();
-        return filterInputText.isEmpty() ? false : startsWithRb.isSelected()
+        return (filterInputText.isEmpty() || propertyValue.isEmpty()) ? false : startsWithRb.isSelected()
                 ? propertyValue.toLowerCase().startsWith(filterInputText) : propertyValue.toLowerCase().contains(filterInputText);
```

Comment:
change this return to `(!filterInputText.isEmpty() && !propertyValue.isEmpty()) || (startsWithRb.isSelected() ? propertyValue.toLowerCase().startsWith(filterInputText) : propertyValue.toLowerCase().contains(filterInputText))`

Reply:
I added original line to avoid further processing when the strings are null. Feels like the above will still try to process when either is null?

Reply:
see above comment

Reply:
Similarly,  && seems to work here.